client/CServerHandler.cpp

```diff
@@ -546,6 +546,11 @@ void CServerHandler::startCampaignScenario(std::shared_ptr<CCampaignState> cs)
 	SDL_PushEvent(&event);
 }
 
+void CServerHandler::showServerError(std::string txt)
```

Comment:
```suggestion
void CServerHandler::showServerError(const std::string & txt)
```

Reply:
That's would be an easy way to get a potential crash in multithreading code

Reply:
tbh I don't understand how (not a multithreading expert). Could you provide an example?

Reply:
I can only imagine such case with async lambdas which also capture variables by &. But this case does not look like a problem of such kind. Also it is more a problem of that async lambdas. They should capture by value.

Reply:
You may have class with std::string datafield, which you can transfer to the function.
Class will be destroyed earlier that function is finished.

Reply:
aha, by multithreading you meant async call basically. And I thought it's about multiple threads calling the method at the same time. The former makes sense.

Reply:
The common approach I prefer is to use copy instead of reference **anytime** when it **may** be asynchronous calls. The cost payed is nothing in comparison with debugging and searching floating multithreading bugs.

Reply:
This looks kind of generic practice and will work only if put to code style like do not pass vars by const & where performance is not critical, pass by value instead.

Reply:
Our code style has nothing regarding passing parameters by const refs btw.

Reply:
Btw we already have void sendMessage(const std::string & txt). Cant we use it?

Reply:
no, this one is for chat message (function name is misleading for sure)